Fixed: does not translate a begin-body when a rescue contains a retry.

mvidner opened the pull request.

mvidner committed:
Fixed: does not translate a begin-body when a rescue contains a retry.

mvidner committed:
Use a container on Travis.

https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments

jreidinger commented on lib/rubocop/yast/track_variable_scope.rb:
```diff
@@ -180,13 +174,20 @@ def on_ensure(node)
+            throw :needle_seen
+          end
+        end
+        needle_seen
```
Hmm, why this throw/catch construct? it looks unnecessary complex, what about
```
def descendant?(node, descendant_type)
  on_node(descendant_type, node) do
    return true
  end

  false
end
```

mvidner replied:
Ah, that is indeed simpler and clearer!

mvidner committed:
Simplified the short circuit.

This is where we actually WANT the non-obvious semantic of "return"
exiting the "def", not only the block.

mvidner committed:
Version 0.0.9
